diff --git a/dist/cli/build/bundle-server.js b/dist/cli/build/bundle-server.js
index e57f6b85a9445e2088b855c36233251ab2a51f16..4d571dae9c904521ef8f4df1ab69d953bdc67c03 100644
--- a/dist/cli/build/bundle-server.js
+++ b/dist/cli/build/bundle-server.js
@@ -71,6 +71,8 @@ export async function bundleServer(buildOpts, projectOpts) {
         minifySyntax: projectOpts.minify && !debug,
         legalComments: "none",
         metafile: true,
+        // Disable keepNames to prevent __name helper that causes runtime errors in Workers
+        keepNames: false,
         // Next traces files using the default conditions from `nft` (`node`, `require`, `import` and `default`)
         //
         // Because we use the `node` platform for this build, the "module" condition is used when no conditions are defined.
